assets/js/components/simulation/VoiceWindow.jsx

```diff
@@ -124,51 +135,37 @@ const VoiceWindow = translate()(
       }, 2000)
     }
 
-    hangUp(): void {
-      this.audio.pause()
-
-      if (this.messageTimer) clearTimeout(this.messageTimer)
-      this.stopRepeatingLastAudio()
-
-      if (!this.props.readOnly) {
-        this.props.onSendMessage({ body: "stop", type: "at" })
+    cancelMessageTimer(): void {
+      if (this.messageTimer) {
+        clearTimeout(this.messageTimer)
+        this.messageTimer = null
       }
     }
 
-    startRepeatingLastAudio(): void {
-      if (!this.audio) return // no last audio available to repeat
-      if (this.repeatMessageTimer) return // already repeating audio
-
-      this.initRepeatTimer(() => {
-        if (this.timesRepeated < 3) {
-          this.clearRepeatTimer()
-          this.repeatLastAudio()
-        } else {
-          this.hangUp()
-        }
-      })
-    }
-
-    stopRepeatingLastAudio(): void {
-      this.clearRepeatTimer()
-      this.timesRepeated = 0
+    stopSimulation(): void {
+      this.audio.pause()
+      this.cancelAnswerTimeout()
+      this.cancelMessageTimer()
     }
 
-    repeatLastAudio(): void {
-      if (!this.audio) return
+    // Simulates a phone hangup by sending a STOP message. This is kinda
+    // hackish, we could probably send a proper hangup message that would be
```

Comment:
Agree. Also because using stop here may be confused with the SMS concept of "stop mo" message. Maybe we could have a new type "conn" and then messages like { body: "hangup", type: "conn" } for this case and { body: "timeout", type: "conn" } or something similar for an unanswered call?

Reply:
I made some failed attempts, but I finally found a mean to implement a proper hangup! Please see my newest commit @aeatencio

Reply:
> we must first locate & understand how are hangups handled in the Verboice channel

I'm not there yet 😞 . But I'm just wondering if this comment still applies 👇 

    // Simulates a phone hangup by sending a STOP message. This is kinda
    // hackish, we could probably send a proper hangup message that would be
    // properly handled by the simulator.